TelepresenceBot/mobile/src/main/java/com/novoda/tpbot/human/HumanActivity.java

```diff
@@ -7,54 +7,62 @@
 import com.novoda.notils.caster.Views;
 import com.novoda.tpbot.Direction;
 import com.novoda.tpbot.R;
-import com.novoda.tpbot.SelfDestructingMessageView;
+import com.novoda.tpbot.support.SelfDestructingMessageView;
+import com.novoda.tpbot.support.SwitchableView;
 
 import java.util.concurrent.TimeUnit;
 
 public class HumanActivity extends AppCompatActivity {
 
     private static final String LAZERS = String.valueOf(Character.toChars(0x1F4A5));
-
     private static final long COMMAND_REPEAT_DELAY = TimeUnit.MILLISECONDS.toMillis(100);
 
-    private Handler handler;
     private SelfDestructingMessageView debugView;
+    private SwitchableView switchableView;
 
-    String currentCommand;
+    private Handler handler;
+    private String currentCommand;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_human);
+
         debugView = Views.findById(this, R.id.bot_controller_debug_view);
+        switchableView = Views.findById(this, R.id.bot_switchable_view);
 
         handler = new Handler();
 
-        ControllerView controllerView = Views.findById(this, R.id.bot_controller_direction_view);
-        controllerView.setControllerListener(new ControllerListener() {
-
-            @Override
-            public void onDirectionPressed(Direction direction) {
-                startRepeatingCommand(direction.visualRepresentation());
-            }
-
-            @Override
-            public void onDirectionReleased(Direction direction) {
-                stopRepeatingCommand(direction.visualRepresentation());
-            }
-
-            @Override
-            public void onLazersFired() {
-                startRepeatingCommand(LAZERS);
-            }
-
-            @Override
-            public void onLazersReleased() {
-                stopRepeatingCommand(LAZERS);
-            }
-        });
+        ControllerView controllerView = Views.findById(switchableView, R.id.bot_controller_direction_view);
+        controllerView.setControllerListener(controllerListener);
+
+        ServerDeclarationView serverDeclarationView = Views.findById(switchableView, R.id.bot_server_declaration_view);
+        serverDeclarationView.setServerDeclarationListener(serverDeclarationListener);
     }
 
+    private final ControllerListener controllerListener = new ControllerListener() {
```

Comment:
I tend to do this for interface declarations because then they are directly below the usage, making the code flow better.